File: catkin_ws/src/ssr_pkg/scripts/yao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

if __name__ == "__main__":
    rospy.init_node("yao_node")
    rospy.logwarn("我是瑶py")

    pub = rospy.Publisher("ge_ge", String, queue_size=10)

    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        rospy.loginfo("11111")
        msg = String()
        msg.data = "求++++"
        pub.publish(msg)
        rate.sleep()

File: catkin_ws/src/ssr_pkg/src/yao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "yao_node");
    printf("过去生于为来\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::String>("ge_ge", 10);

    ros::Rate loop_rate(10);

    while (ros::ok())
    {
        printf("2222\n");
        std_msgs::String msg;
        msg.data = "求++++";
        pub.publish(msg);
        loop_rate.sleep();
    }
    return 0;
}
